mongo/mongo.go

```diff
@@ -280,8 +280,16 @@ func SelectPeerAddress(addrs []network.Address) string {
 // SelectPeerHostPort returns the HostPort to use as the mongo replica set peer
 // by selecting it from the given hostPorts.
 func SelectPeerHostPort(hostPorts []network.HostPort) string {
+	// TODO(macgreagoir) IPv6. We were always choosing a cloud-local or
+	// localhost addr here (with true arg) but this doesn't make sense in
+	// IPv6, where the IPv6 [public] address is ignored in favour of
+	// ip6-localhost. Only pass true if we find an IPv4 address in the
+	// slice.
 	logger.Debugf("selecting mongo peer hostPort by scope from %+v", hostPorts)
-	return network.SelectMongoHostPortsByScope(hostPorts, true)[0]
+	return network.SelectMongoHostPortsByScope(
+		hostPorts,
+		network.HostPortWithIPv4Address(hostPorts),
```

Comment:
can you use a local variable so that the meaning of this makes more sense? When I read it at a glance it looks like we are only selecting IPv4 addresses.

allowMachineLocal := network.HostPortWithIPv4Address(hostPorts)

return network.SelectMongoHostPortsByScope(hostPorts, allowMachineLocal)

I'm still not 100% sold on using machine local. This feels more like "if we don't *have* a cloud-local then maybe we can fall back to localhost".
Given its the address we are putting into the replicaSet, it seems like it can't ever be right to give a link local address if you are ever actually in a peer group.